src/index.js

```diff
@@ -119,8 +120,18 @@ function(err, results) {
     shell.exec('npm version --no-git-tag-version ' + oldVersion);
   }
 
-  // ### STEP 8 - Run if any pre commit script has been specified (DESTRUCTIVE OPERATION)
+  // ### STEP 8 - Run if any pre commit script has been specified (DESTRUCTIVE OPERATION?)
   lib.runPreCommitScript(program.preCommit, version);
+
   // ### STEP 9 - Tag and push (DESTRUCTIVE OPERATION)
-  if (!program.dryrun) lib.addFilesAndCreateTag(version);
+  let pushResultCode = 1; // Default to failure signal
+
+  if (!program.dryrun) {
+    pushResultCode = lib.addFilesAndCreateTag(version);
+  }
+
+  // ### STEP 10 - Run after successful push (DESTRUCTIVE OPERATION?)
+  if (pushResultCode === 0 || program.dryrun) {
```

Comment:
Yes, it is a hard one. can you check the output log to contain ``>>> about to run your "post-success" script called`. It doesn't really matter the outcome of `npm run ${program.postSuccess}`, as long as it gets called, the test case passes in my opinion. Do you think this is reasonable for this test case?